runtime-watcher/internal/handler.go

```diff
@@ -34,17 +33,17 @@ const (
 )
 
 type Handler struct {
-	Client     client.Client
-	Logger     logr.Logger
-	Parameters ServerParameters
+	Client       client.Client
+	Logger       logr.Logger
+	Parameters   ServerParameters
+	Deserializer runtime.Decoder
 }
 
 type ServerParameters struct {
 	Port        int    // webhook server port
 	CACert      string // CA key used to sign the certificate
 	TLSCert     string // path to TLS certificate for https
 	TLSKey      string // path to TLS key matching for certificate
-	TLSServer   bool   // indicates if an HTTPS server should be created
 	TLSCallback bool   // indicates if KCP accepts HTTP or HTTPS requests
```

Comment:
Is there a reason you didn't remove the TLSCallback as well?

Reply:
Of course, removed.

Reply:
Seems it was introduced to make testing easier and bypass https for the test. Will try to introduce cert generation in the test

Reply:
As discussed we will enable tls for the env test in a separate [issue](https://github.com/kyma-project/runtime-watcher/issues/129).